PencilCase/PencilCase/scripts/Navigator.ts

```diff
@@ -36,83 +37,103 @@ export class Navigator {
 
     public goHome = () => {
         Application.instance.activePage(new HomePage());
-        $(':mobile-pagecontainer').pagecontainer("change", "#HomePage");
+        $(':mobile-pagecontainer').pagecontainer("change", "#" + Consts.Pages.HomePage.Id, { showLoadMsg: true });
     }
 
+    public showConfirmDialog(header: string, content: string, confirm: () => void, cancel?: () => void) {
```

Comment:
Added a application level confirm dialog (actually a page with data-role set as pop up) to display one confirming dialog with specified header and messages, and specified call backs.